src/service/debtor-service.ts

```diff
@@ -0,0 +1,303 @@
+ * @param balance
+ */
+function calculateFine(balance: DineroObject | DineroObjectResponse | DineroObjectRequest): Dinero {
+  return DineroTransformer.Instance.from(
+    Math.max(
+      Math.min(
+      // Divide by 5, round to euros (/100, then floor), then multiply by 100 again
+        Math.floor(-balance.amount / 500) * 100,
+        500,
+      ),
+      0,
+    ),
+  );
+}
+
+export default class DebtorService {
+  static asFineResponse(fine: Fine): FineResponse {
+    return {
+      id: fine.id,
+      createdAt: fine.createdAt.toISOString(),
+      updatedAt: fine.updatedAt.toISOString(),
+      user: parseUserToBaseResponse(fine.userFineGroup.user, false),
+      amount: {
+        amount: fine.amount.getAmount(),
+        precision: fine.amount.getPrecision(),
+        currency: fine.amount.getCurrency(),
+      },
+    };
+  }
+
+  static asBaseFineHandoutEventResponse(e: FineHandoutEvent): BaseFineHandoutEventResponse {
+    return {
+      id: e.id,
+      createdAt: e.createdAt.toISOString(),
+      updatedAt: e.updatedAt.toISOString(),
+      referenceDate: e.referenceDate.toISOString(),
+      createdBy: parseUserToBaseResponse(e.createdBy, false),
+    };
+  }
+
+  static asFineHandoutEventResponse(e: FineHandoutEvent): FineHandoutEventResponse {
+    return {
+      ...this.asBaseFineHandoutEventResponse(e),
+      fines: e.fines.map((fine) => this.asFineResponse(fine)),
+    };
+  }
+
+  /**
+   * Get a list of all fine handout events in chronological order
+   */
+  public static async getFineHandoutEvents(pagination: PaginationParameters = {}): Promise<PaginatedFineHandoutEventResponse> {
+    const { take, skip } = pagination;
+
+    const events = await FineHandoutEvent.find({ take, skip });
+    const count = await FineHandoutEvent.count();
+
+    const records = events.map((e) => DebtorService.asBaseFineHandoutEventResponse(e));
+
+    return {
+      _pagination: {
+        take, skip, count,
+      },
+      records,
+    };
+  }
+
+  /**
+   * Return the fine handout event with the given id. Includes all its fines with the corresponding user
+   */
+  public static async getSingleFineHandoutEvent(id: number): Promise<FineHandoutEventResponse> {
+    const fineHandoutEvent = await FineHandoutEvent.findOne({
+      where: { id },
+      relations: ['fines', 'fines.userFineGroup', 'fines.userFineGroup.user'],
+      order: { createdAt: 'DESC' },
+    });
+
+    return DebtorService.asFineHandoutEventResponse(fineHandoutEvent);
+  }
+
+  /**
+   * Return all users that had at most -5 euros balance both now and on the reference date
+   * For all these users, also return their fine based on the reference date.
+   * @param userTypes List of all user types fines should be calculated for
+   * @param referenceDate Date to base fines on. If undefined, use now.
+   */
+  public static async calculateFinesOnDate({ userTypes, referenceDate }: CalculateFinesParams): Promise<UserToFineResponse[]> {
```

Comment:
klopt deze jsdoc? is het niet een calculateFinesParams object als param?

Reply:
Ja dit mag! Zie screenshot
![image](https://github.com/GEWIS/sudosos-backend/assets/38660000/6dcabdd4-c25d-433f-a009-d78dc1da75f4)